File: field/state.go
package field

import (
	"fmt"
)

type State int

const (
	Unknown  State = iota
	Flagged  State = iota
	Revealed State = iota
)

// String returns a text version of the State enum.
func (s State) String() string {
	switch s {
	case Unknown:
		return "unknown"
	case Flagged:
		return "flagged"
	case Revealed:
		return "revealed"
	}
	panic(fmt.Sprintf("invalid state %d", s))
}

File: src/field/state.rs
use std::fmt;

#[derive(Debug, Clone, Copy, PartialEq, Eq)]
pub enum State {
    Unknown,
    Flagged,
    Revealed,
}

impl fmt::Display for State {
    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
        match self {
            State::Unknown => write!(f, "unknown"),
            State::Flagged => write!(f, "flagged"),
            State::Revealed => write!(f, "revealed"),
        }
    }
}
